javascript/tagAutocomplete.js

```diff
@@ -666,6 +670,30 @@ function addResultsToList(textArea, results, tagword, resetList) {
     let tagColors = TAC_CFG.colorMap;
     let mode = (document.querySelector(".dark") || gradioApp().querySelector(".dark")) ? 0 : 1;
     let nextLength = Math.min(results.length, resultCount + TAC_CFG.resultStepLength);
+    const IS_DAN_OR_E621_TAG_FILE =  (tagFileName.toLowerCase().startsWith("danbooru") || tagFileName.toLowerCase().startsWith("e621"))
+
+    let prompt = "", tagCount = {}
+
+    // Indicate if has existed tag before 
+    if(IS_DAN_OR_E621_TAG_FILE) {
+        prompt = textArea.value.trim()
```

Comment:
Not a fan of this double variable initialization. I'd suggest moving `prompt` into the if scope below as you only use it there anyway.

```suggestion
    const tagCount = {};

    // Indicate if tag was used before 
    if (IS_DAN_OR_E621_TAG_FILE) {
        const prompt = textArea.value.trim();
```